### PR TITLE
BUG: ensure proper metadata for 3D geometry types in to_parquet

### LINKED ISSUE #2824: BUG: to_parquet doesn't set correct geometry_type in GeoParquet metadata for 3D geometries
The geometry_type should be "POINT Z" and not just "POINT":

```
In [12]: df = geopandas.GeoDataFrame({'col': [1, 2, 3]}, geometry=geopandas.points_from_xy([1, 2, 3], [2, 3, 4], [1, 1, 1]))

In [13]: df
Out[13]: 
   col                           geometry
0    1  POINT Z (1.00000 2.00000 1.00000)
1    2  POINT Z (2.00000 3.00000 1.00000)
2    3  POINT Z (3.00000 4.00000 1.00000)

In [14]: df.to_parquet("test_z_dimension.parquet")

In [15]: import pyarrow.parquet as pq

In [16]: meta = pq.read_metadata("test_z_dimension.parquet")

In [19]: meta.metadata[b"geo"]
Out[19]: b'{"primary_column": "geometry", "columns": {"geometry": {"encoding": "WKB", "crs": null, "geometry_types": ["Point"], "bbox": [1.0, 2.0, 3.0, 4.0]}}, "version": "1.0.0-beta.1", "creator": {"library": "geopandas", "version": "0.12.1+62.ga24aad54"}}'
```

### PR DESCRIPTION
**Pull request by JohnMoutafis**
Attempts to fix `.to_parquet` not respecting "Z" geometries.

Closes #2824

### CODE REVIEW

**Comment by JohnMoutafis** [comment]
Hadn't setup the pre commit hook so I had to make some linding connections 😓

**Comment by JohnMoutafis** [comment]
@martinfleis `Tests / ubuntu-latest, ci/envs/310-latest-conda-forge.yaml (pull_request)` seems to be stuck. Can we rerun it?

File: geopandas/io/arrow.py [hunk 1]
```diff
@@ -94,7 +94,25 @@ def _create_metadata(df, schema_version=None):
...
+        # https://github.com/opengeospatial/geoparquet/blob/main/format-specs/geoparquet.md#geometry_types
+        # Instead of going through the `_vectorized.geometry_type_values`
+        # that do not include the "Z" varieties.
+        geometry_types = sorted(
+            Series(
+                list(
+                    set(
+                        [
+                            f"{geom} Z" if has_z else geom
+                            for geom, has_z in zip(series.geom_type, series.has_z)
```
Comment (jorisvandenbossche) [inline geopandas/io/arrow.py:108]: Could you maybe write this in two separate steps? (eg first creating the set, and then the `sorted(Series(list(..` on a next line) That should help with the formatting here

**Review by jorisvandenbossche — COMMENTED** [review]
@JohnMoutafis Thanks for the PR!

We will also want to test this. You can take a look at geopandas/io/tests/test_arrow.py, for example add a new test based on `def test_create_metadata`, but then one that just asserts the "geometry_type" part of the metadata for some varying input.

File: geopandas/io/arrow.py (hunk 1 above)
Comment (JohnMoutafis) [reply geopandas/io/arrow.py:108]: Sorry for the late reply! Ok this is easily fixed, I am unsure on how to proceed with the testing. Looking into it!

Comment (JohnMoutafis) [reply geopandas/io/arrow.py]: Fixed in the last commit

**Comment by JohnMoutafis** [comment]
@jorisvandenbossche it doesn't seem that the test fail is related to my changes

**Comment by m-richards** [comment]
> @jorisvandenbossche it doesn't seem that the last test fail is related to my changes

That looks right, nothing to be worried about in this PR (the failing environment is built against the nightly dev release of pandas, so the failures are due to upstream changes we'll need to incorporate)

**Comment by JohnMoutafis** [comment]
@jorisvandenbossche any news on that?

**Comment by jorisvandenbossche** [comment]
@JohnMoutafis apologies for the very slow follow-up. I rebased this again to fix the conflicts with latest main, cleaned up a few comments and unnecessary test code, but for the rest it is looking good! Thanks again for the contribution

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**